pandas/core/frame.py

```diff
@@ -5545,20 +5545,46 @@ def _replace_columnwise(
     @doc(NDFrame.shift, klass=_shared_doc_kwargs["klass"])
     def shift(
         self,
-        periods: int = 1,
+        periods: int | Iterable[int] = 1,
         freq: Frequency | None = None,
         axis: Axis = 0,
         fill_value: Hashable = lib.no_default,
+        suffix: str | None = None,
     ) -> DataFrame:
-        axis = self._get_axis_number(axis)
-
         if freq is not None and fill_value is not lib.no_default:
             # GH#53832
             raise ValueError(
                 "Cannot pass both 'freq' and 'fill_value' to "
                 f"{type(self).__name__}.shift"
             )
 
+        axis = self._get_axis_number(axis)
+
+        if is_list_like(periods):
+            # periods is not necessarily a list, but otherwise mypy complains.
+            periods = cast(list, periods)
```

Comment:
Does mypy complain if you cast to Iterable?

Reply:
Actually yes, because we're doing `len` later. But that probably means the type of periods shouldn't be `Iterable | int`, but `Sequence | int`.